app/src/main/java/com/allan/boardbuddies/activities/CanvasActivity.java

```diff
@@ -120,15 +128,18 @@ public void onFragmentResult(@NonNull String requestKey, @NonNull Bundle bundle)
     public boolean onNavigationItemSelected(@NonNull MenuItem item) {
         switch (item.getItemId()){
             case R.id.add_text:
-                showDialog();
+                showDialog("add_text");
```

Comment:
Ok, I understand you're passing this in to use as the resultKey. But as I explained in previous comment, you shouldn't be using resultKey to distinguish between the use cases. So, given that you remove that part...

To be honest I think it makes more sense to create a whole separate fragment for editing the title. If you think about it, later on your captions text editor fragment will want to support other things like font, size, colour, etc. But for title, all you'll need is to set the text. Instead of using a hardcoded string to determine what the fragment does, it's better to do the separation of logic in the code structure itself.

For now, if you don't want to make a new fragment, I would recommend just using a boolean flag "isEditingTitle" in TextDialogFragment.newInstance() to distinguish between daption vs title